node/src/reactor.rs

```diff
@@ -338,7 +342,8 @@ where
         let inner_enter = event_span.enter();
 
         // We log events twice, once in display and once in debug mode.
-        debug!(%event, ?q);
+        let event_as_string = format!("{}", event);
+        debug!(event=%event_as_string, ?q);
```

Comment:
This was fine I think, no need to change.

Reply:
Ah, I see now. You need it to be of string type to use in the later `debug!` call.

Reply:
It's more about keeping something around which can be used in the `warn!` call below, since the `event` is consumed before then.

Since the event itself can't (and probably shouldn't) be cloned, this is just to keep a copy of the Display impl around for later.